src/manager.cpp

```diff
@@ -93,6 +93,121 @@ uint32_t SessionManager::closeAllByType(SessionType type)
+            if ((allowed) && !(ownedOnly))
+            {
+                return session->sessionType() == type && id != numSessId;
+            }
+            else
+            {
+                return session->sessionType() == type &&
+                       session->getOwner() == userName && id != numSessId;
+            }
+        });
+    return count;
+}
+
+bool SessionManager::isAllSessionsAllowed(std::string sessionId)
+{
+    log<level::DEBUG>("SessionManager::isAllSessionsAllowed()",
+                      entry("CALLER-SESSION-ID=%s", sessionId.c_str()));
+
+    std::map<std::string,
+             std::variant<std::string, std::vector<std::string>, bool>>
+        mapperResponse;
+    auto mapper = bus.new_method_call(
+        "xyz.openbmc_project.User.Manager", "/xyz/openbmc_project/user",
+        "xyz.openbmc_project.User.Manager", "GetUserInfo");
+
+    std::string userName = getSessionItem(sessionId)->getOwner();
+    mapper.append(userName.c_str());
+
+    auto mapperResponseMsg = bus.call(mapper);
+
+    if (mapperResponseMsg.is_method_error())
+    {
+        log<level::ERR>("Error getting user info",
+                        entry("SERVICE=%s", "xyz.openbmc_project.User.Manager"),
+                        entry("USERNAME=%s", userName.c_str()));
+        throw InternalFailure();
+    }
+
+    mapperResponseMsg.read(mapperResponse);
+
+    if (mapperResponse.empty())
+    {
+        log<level::ERR>("No Object has implemented the interface",
+                        entry("SERVICE=%s", "xyz.openbmc_project.User.Manager"),
+                        entry("USERNAME=%s", userName.c_str()));
+        elog<InternalFailure>();
+        throw InternalFailure();
+    }
+
+    auto mapperResponseIt = mapperResponse.find("UserPrivilege");
+    if (mapperResponseIt == mapperResponse.end())
+    {
+        throw InternalFailure();
+    }
+    std::string privilegeValue = std::get<0>(mapperResponseIt->second);
+    return (privilegeValue == "priv-admin");
+}
+
+// required to follow DRY principle, since logic would be used several times
+SessionItemPtr SessionManager::getSessionItem(std::string sessionId)
+{
+    log<level::DEBUG>("SessionManager::getSessionItem()",
+                      entry("CALLER-SESSION-ID=%s", sessionId.c_str()));
+
+    auto numSessId = parseSessionId(sessionId);
+    auto foundSessionIt = sessionItems.find(numSessId);
+    if (foundSessionIt == sessionItems.end())
+    {
+        throw InvalidArgument();
+    }
+    return foundSessionIt->second;
+}
+
+bool SessionManager::isOwnSession(std::string callerSessionId,
+                                  std::string removedSessionId)
+{
+    log<level::DEBUG>("SessionManager::isOwnSession()",
+                      entry("CALLER-SESSION-ID=%s", callerSessionId.c_str()),
+                      entry("REMOVED-SESSION-ID=%s", removedSessionId.c_str()));
+
+    std::string userName1 = getSessionItem(callerSessionId)->getOwner();
```

Comment:
`const` of both `userName`. 
And the var name could be more clear, e.g. `caller UserName` and `removed UserName`